src/static/templateUrl.ts

```diff
@@ -49,16 +51,16 @@ export default class TemplateURL extends URL {
    * are allowed and are appended using the "repeat" convention where the \{ a:
    * ["1", "2"] \} becomes "?a=1&a=2"
    */
-  addQueryParams(queryParams: {[key: string]: unknown} | undefined): void {
+  addQueryParams(queryParams?: QueryParameters): void {
     if (queryParams) {
       Object.keys(queryParams).forEach((key) => {
-        if (Array.isArray(queryParams[key])) {
-          const paramArr = queryParams[key] as unknown[];
-          for (let i = 0; i < paramArr.length; i += 1) {
-            this.searchParams.append(key, String(paramArr[i]));
+        const param = queryParams[key];
+        if (Array.isArray(param)) {
+          for (let i = 0; i < param.length; i += 1) {
```

Comment:
I have been told it is more performant to move the .length to a variable outside the for loop definition.